cxx/pclean/pclean.cc

```diff
@@ -8,12 +8,49 @@
 #include <random>
 
 #include "cxxopts.hpp"
+#include "irm.hh"
 #include "hirm.hh"
 #include "inference.hh"
+#include "util_io.hh"
+#include "pclean/csv.hh"
 #include "pclean/io.hh"
 #include "pclean/schema.hh"
 #include "pclean/schema_helper.hh"
 
+T_observations translate_observations(
+    const DataFrame& df, const T_schema &schema) {
+  T_observations obs;
+
+  for (const auto& col : df.data) {
+    const std::string& col_name = col.first;
+    const T_relation& trel = schema.at(col_name);
+    size_t num_domains;
```

Comment:
Optional: IMO this is more concise and a little more readable than closing over `num_domains`

```
num_domains = std::visit([](const auto &r) {return r.domains.size();}, trel);
```

Reply:
Done.